// @material-tailwind/react/dist/theme/checkbox.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/checkbox.ts
var checkbox_exports = {};
__export(checkbox_exports, {
  checkboxIndicatorTheme: () => checkboxIndicatorTheme,
  checkboxTheme: () => checkboxTheme
});
module.exports = __toCommonJS(checkbox_exports);
var checkboxTheme = {
  defaultProps: {
    color: "primary"
  },
  baseStyle: "group shadow-sm shadow-black/5 inline-block relative h-5 w-5 cursor-pointer rounded bg-transparent border border-surface transition-all duration-200 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none hover:shadow-md",
  color: {
    primary: "data-[checked=true]:bg-primary data-[checked=true]:border-primary text-primary-foreground",
    secondary: "data-[checked=true]:bg-secondary data-[checked=true]:border-secondary text-secondary-foreground",
    info: "data-[checked=true]:bg-info data-[checked=true]:border-info text-info-foreground",
    success: "data-[checked=true]:bg-success data-[checked=true]:border-success text-success-foreground",
    warning: "data-[checked=true]:bg-warning data-[checked=true]:border-warning text-warning-foreground",
    error: "data-[checked=true]:bg-error data-[checked=true]:border-error text-error-foreground"
  }
};
var checkboxIndicatorTheme = {
  baseStyle: "pointer-events-none absolute left-2/4 top-2/4 text-current -translate-x-2/4 -translate-y-2/4 scale-75 opacity-0 transition-all duration-200 ease-in data-[checked=true]:scale-100 data-[checked=true]:opacity-100"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  checkboxIndicatorTheme,
  checkboxTheme
});
