// @material-tailwind/react/dist/components/tabs.js
"use client";
import {
  Tabs,
  TabsList,
  TabsPanel,
  TabsRoot,
  TabsTrigger,
  TabsTriggerIndicator,
  tabs_default
} from "../chunk-CATVNKHO.js";
import "../chunk-JVDTBYWX.js";
import "../chunk-JZXXEFZT.js";
import "../chunk-GUZBX522.js";
import "../chunk-CXJEJBKT.js";
import "../chunk-CPR7EA3T.js";
import "../chunk-CBF6OFZW.js";
import "../chunk-7SWBG6UK.js";
import "../chunk-FHF6P4UN.js";
import "../chunk-APTGGM6D.js";
import "../chunk-BAFDZUY3.js";
import "../chunk-AGIFMRNV.js";
import "../chunk-77ZNGTSH.js";
import "../chunk-5VIFYY3R.js";
import "../chunk-C2SK4YPI.js";
import "../chunk-QTMDCPR3.js";
import "../chunk-U44YIC2J.js";
import "../chunk-QHXOBAAC.js";
import "../chunk-BGYX7VND.js";
import "../chunk-BGJMALSI.js";
import "../chunk-YVCLQOWL.js";
import "../chunk-332RUHED.js";
import "../chunk-4JMI2FIT.js";
import "../chunk-BQYKGMWX.js";
import "../chunk-ANNCLIQH.js";
import "../chunk-LRNZTCBT.js";
import "../chunk-ADIUZG7W.js";
import "../chunk-C4GS2FOC.js";
import "../chunk-YTQ4A2UH.js";
import "../chunk-PTC7KVHU.js";
import "../chunk-V2YBIGWT.js";
import "../chunk-FOAJJTIZ.js";
import "../chunk-LN4CMBOV.js";
import "../chunk-VYISNWQG.js";
import "../chunk-FQY26STY.js";
import "../chunk-XWS4SIB5.js";
import "../chunk-5DV345YV.js";
import "../chunk-L3VNOE5J.js";
import "../chunk-AXCKF2IS.js";
export {
  Tabs,
  TabsList,
  TabsPanel,
  TabsRoot,
  TabsTrigger,
  TabsTriggerIndicator,
  tabs_default as default
};
